lib/msal-node/src/cache/Storage.ts

```diff
@@ -194,8 +194,9 @@ export class Storage extends CacheManager {
      */
     getIdTokenCredential(idTokenKey: string): IdTokenEntity | null {
         const credType = CredentialEntity.getCredentialType(idTokenKey);
-        if (credType === CredentialType.ID_TOKEN) {
-            return this.getItem(idTokenKey) as IdTokenEntity;
+        const idToken = this.getItem(idTokenKey) as IdTokenEntity;
+        if (IdTokenEntity.isIdTokenEntity(idToken) && credType === CredentialType.ID_TOKEN) {
```

Comment:
credType check is part of validator
```suggestion
        if (idToken && IdTokenEntity.isIdTokenEntity(idToken)) {
```

Reply:
Actually we wouldn't need `idToken &&` at all as `IdTokenEntity.isIdTokenEntity` takes care of an undefined entity. Let me make those changes. And yes, `CredentialType` check is also covered there.